provider/pkg/await/service.go

```diff
@@ -120,32 +122,34 @@ func (sia *serviceInitAwaiter) Await() error {
 	//   4. External IP address is allocated (if we're type `LoadBalancer`).
 	//
 
-	serviceClient, endpointsClient, err := sia.makeClients()
-	if err != nil {
-		return err
+	stopper := make(chan struct{})
+	defer close(stopper)
+
+	namespace := sia.config.currentInputs.GetNamespace()
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
 	}
+	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(sia.config.clientSet.GenericClient, 60*time.Second, namespace, nil)
+	informerFactory.Start(stopper)
 
-	// Create service watcher.
-	serviceWatcher, err := serviceClient.Watch(context.TODO(), metav1.ListOptions{})
+	serviceEvents := make(chan watch.Event)
+	serviceInformer, err := informers.New(informerFactory, informers.ForServices(), informers.WithEventChannel(serviceEvents))
 	if err != nil {
-		return errors.Wrapf(err, "Could set up watch for Service object '%s'",
-			sia.config.currentInputs.GetName())
+		return err
 	}
-	defer serviceWatcher.Stop()
+	go serviceInformer.Informer().Run(stopper)
 
-	// Create endpoint watcher.
-	endpointWatcher, err := endpointsClient.Watch(context.TODO(), metav1.ListOptions{})
+	endpointsEvents := make(chan watch.Event)
+	endpointsInformer, err := informers.New(informerFactory, informers.ForEndpoints(), informers.WithEventChannel(endpointsEvents))
 	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for Endpoint objects associated with Service %q",
-			sia.config.currentInputs.GetName())
+		return err
 	}
-	defer endpointWatcher.Stop()
+	go endpointsInformer.Informer().Run(stopper)
 
 	version := cluster.TryGetServerVersion(sia.config.clientSet.DiscoveryClientCached)
 
 	timeout := metadata.TimeoutDuration(sia.config.timeout, sia.config.currentInputs, DefaultServiceTimeoutMins*60)
-	return sia.await(serviceWatcher, endpointWatcher, time.After(timeout), make(chan struct{}), version)
+	return sia.await(serviceEvents, endpointsEvents, time.After(timeout), make(chan struct{}), version)
 }
 
 func (sia *serviceInitAwaiter) Read() error {
```

Comment:
I noticed that the Job awaiter Read method was updated to use the Informer.Logger. Any reason to prefer the old way over that?

Reply:
The job awaiter's Read uses PodAggregator which I thought would be a good candidate for using a lister backed by the informer anyway (as part of the corresponding watch). As a result, I switched Job awaiter's Read to use an informer as well. For the other awaiters I didn't see any advantage to switching Read to informers so let that be.